### ISSUE TITLE
Boolean comparison does not use type guard to narrow

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/TypeScript#32956]
**TypeScript Version:**  3.5.2.

**Search Terms:** type guard narrow if statement ternary condition false true strict equality

**Code**

```ts
// This works
const fn1 = (value: unknown) =>
    !Array.isArray(value) ? undefined : value[0];

// This doesn't—it should be the same as above?
const fn2 = (value: unknown) =>
    Array.isArray(value) === false
        ? undefined
        : // Unexpected error: Object is of type 'unknown'.
            value[0];
```

**Comment by AnyhowStep** [comment]
Duplicate,
https://github.com/microsoft/TypeScript/issues/32819
